client/x/evmstaking/types/withdraw_test.go

```diff
@@ -84,6 +85,11 @@ func (suite *WithdrawTestSuite) TestWithdrawalsString() {
 		suite.delAddr, suite.valAddr, suite.evmAddr.String(),
 	)
 	require.Equal(expectedString, ws.String())
+
+	// Test with spaces, it should trim the spaces
+	stringWithSpaces := " " + ws.String() + " " // add leading and trailing spaces
+	trimmedString := strings.TrimSpace(stringWithSpaces)
```

Comment:
I think we want to test that `String()` will correctly trim spaces not `TrimSpace` can trim spaces. You should add a new test case where it has leading or trailing spaces in the resulted string from a `NewWithdrawal`. If it is impossible to construct such a case, you may skip the test case.

Reply:
@limengformal 
I think it's good to skip the test case because testing that case is quite tricky. 
To test that part, we would need to mock the `String()` that intentionally returns a string with leading or trailing spaces. However, this seems quite complex and cumbersome.

Reply:
@zsystm Then let's remove it

Reply:
@limengformal 
You mean let's remove the code we are currently discussing in the comments? 
I'll remove it soon.

Reply:
yes